src/promptflow-evals/promptflow/evals/evaluate/_eval_run.py

```diff
@@ -314,7 +341,8 @@ def log_artifact(self, artifact_folder: str, artifact_name: str = EVALUATION_ART
         :param artifact_folder: The folder with artifacts to be uploaded.
         :type artifact_folder: str
         """
-        if self._is_broken:
+        self._raise_not_started_nay_be()

```

Comment:
Do we really need this check in all the log methods? Shouldn't we just print a warning saying something like, "Failed to create the run; run artifacts won't be uploaded. Please try again later."

Reply:
This check is added for developers to avoid setting wrong status and mishandling of the run. Customer should not see this issue, as this is an internal class, however, if it happens our unit tests MUST fail.

Reply:
What's the point of showing a developer-facing message to the customer? This could cause confusion. If the run fails to be created in _start_run(), let the subsequent calls fail instead. Wouldn't that be better?

Reply:
In this case should be marked as broken, by our code the run should never be in Non-started state. This exception is added for code integrity, in case we will occasionally change this logic. If this issue will be only logged, we may not be able to catch it by the unit tests and customer will only not get the traces and will find out the error message in the logs. This error is primarily for us to avoid this state.